server/auth/auth.go

```diff
@@ -1348,7 +1347,7 @@ func (auth *AuthManager) handleMatchStatus(conn comms.Link, msg *msgjson.Message
 			"cannot use route 'match_status' on an unauthorized connection")
 	}
 	var matchReqs []msgjson.MatchRequest
-	err := json.Unmarshal(msg.Payload, &matchReqs)
+	err := msg.Unmarshal(&matchReqs)
```

Comment:
`nil` slice is no trouble for `handleMatchStatus` and `handleOrderStatus` afaict

Reply:
But it's not sensical, so should be an error, no?

Reply:
Pushed https://github.com/decred/dcrdex/pull/1014/commits/bdf63b8ea94ad0c9a794fc5e9722db60d3009d2a.
`handleOrderStatus` caught it already with a len check, and I documented the intent there.
But `handleMatchStatus` currently is happy to issue a positive response of `[]` for a payload of `[]`, so I didn't want to change that behavior.  I did however make `null` payload an error now, and documented the current behavior.